kasa/smart/smartdevice.py

```diff
@@ -84,11 +84,19 @@ async def _initialize_children(self):
     @property
     def children(self) -> Sequence[SmartDevice]:
         """Return list of children."""
-        # Wall switches with children report all modules on the parent only
-        if self.device_type == DeviceType.WallSwitch:
-            return []
         return list(self._children.values())
 
+    @property
+    def modules(self) -> dict[str, SmartModule]:
+        """Return the device modules."""
+        if self._device_type == DeviceType.WallSwitch and self._children:
+            modules = {k: v for k, v in self._modules.items()}
+            for child in self._children.values():
+                for modname, mod in child._modules.items():
+                    modules[modname] = mod
```

Comment:
This initialization should probably be done only once?

Reply:
That's done